app/controllers/gifts_controller.rb

```diff
@@ -1,50 +1,45 @@
 class GiftsController < ApplicationController
-  before_action :set_gift, only: %i[show edit update destroy]
   before_action :set_user, except: %i[destroy]
```

Comment:
You should do it in the same style, so if you used memoization for the gift, use it for the user too